### DISCUSSION TITLE
[FEATURE REQUEST] The way to update code for iOS app by Air,like Android?

### DISCUSSION THREAD

**Discussion by AdamSun** [airsdk/Adobe-Runtime-Support#607] [Q&A]
Is there a way to download and update the code to fix bug for ios app without release and upload a new package.It works in Android,but Apple forbid it.Do you get some way to achieve the goal?

**Comment by marchbold** [comment]
Probably more of a forum question than a bug? Apple don't like seeing apps use this approach and will more often than not make your application fail review.

**Comment by kramsurfer** [comment]
It's 'forbid' by the Apple agreement.  All code must be in the package delivered for review by the apple team.  This is a security thing.  Zero code allowed in your .swf's.  

Even constants defined in AS3 in downloaded content have failed review for me.  I had to create 'visual elements' in the swfs to act as constants.

**Comment by marcanw** [comment]
Well I'm kind of an edge case but I do update my code on regular basis (android and ios).
All my screens are webviews loading html code on our server. So I often update the html code  for fixing bugs or glitches.
One day after a failed review on iOS about this particular matter, I talked with the reviewer and he finally told me that if the functionalities and the UI/UX was the same it was fine.
So go figure, maybe html is considered as "content" otherwise they might have problems with all the applications that display dynamic html pages...
Anyway don't quote me to Apple please ;-)

**Comment by rewb0rn** [comment]
One thing that would be great though is if .swf files downloaded from a server would keep the export names of the MovieClips. Currently they are being removed from the .swf file because they are considered "code". It would be nice if we could download a .swf from our server, and then create instances of the MovieClip with the export name "MyMovieClip" for example. 

Currently we have to take the instance from the stage and then move it somewhere else. It works but I don't know if there is any way to create more than one instance of it.
